JitPack Downloads Service Test Failing

:clock11: **When did the problem start?**
~March 2nd

<!-- Indicate when the problem started -->

:camera: **Live badge**

<!-- Provide a link to the live badge in plain text and markdown.  -->
https://img.shields.io/jitpack/dm/github/jitpack/maven-simple
![](https://img.shields.io/jitpack/dm/github/jitpack/maven-simple)

:wrench: **Is the live badge working?**
Doesn't look like it, seeing `invalid response data` error consistently
<!-- Indicate whether or not the live badge is working. -->

:link: **CircleCI link**
https://circleci.com/gh/badges/daily-tests/961
<!-- Provide a link to the failing test in CircleCI. -->

:beetle: **Stack trace**

```
1. JitpackDownloads [live] monthly (github) [ GET /dm/github/dcendents/android-maven-gradle-plugin.json ] 
  ValidationError: message mismatch: "value" with value "invalid response data" fails to match the required pattern: /^([1-9][0-9]*[kMGTPEZY]?|[1-9]\.[1-9][kMGTPEZY])\/(year|month|four weeks|week|day)$/

2. JitpackDownloads [live] unknown package (github) [ GET /dw/github/some-bogus-user/super-fake-project.json ]
  AssertionError: message mismatch: expected 'invalid response data' to equal '0/week'

3. JitpackDownloads [live] weekly (github) [ GET /dw/github/jitpack/maven-simple.json ] 
  ValidationError: message mismatch: "value" with value "invalid response data" fails to match the required pattern: /^([1-9][0-9]*[kMGTPEZY]?|[1-9]\.[1-9][kMGTPEZY])\/(year|month|four weeks|week|day)$/
```

:bulb: **Possible solution**
Not sure yet, guessing the upstream API contract has been changed and the response is missing or has renamed a field we need and have required in the schema
<!--- Optional: only if you have suggestions on a fix/reason for the bug -->

<!-- Love Shields? Please consider donating $10 to sustain our activities:
👉  https://opencollective.com/shields -->

--- a/services/jitpack/jitpack-downloads.service.js
+++ b/services/jitpack/jitpack-downloads.service.js
@@ -1,83 +1,13 @@
 'use strict'
 
-const Joi = require('@hapi/joi')
-const { nonNegativeInteger } = require('../validators')
-const { downloadCount } = require('../color-formatters')
-const { metric } = require('../text-formatters')
-const { BaseJsonService } = require('..')
+const { deprecatedService } = require('..')
 
-const schema = Joi.object({
-  week: nonNegativeInteger,
-  month: nonNegativeInteger,
-}).required()
-
-const intervalMap = {
-  dw: {
-    api_field: 'week',
-    suffix: '/week',
+module.exports = deprecatedService({
+  route: {
+    base: 'jitpack',
+    pattern: ':interval(dw|dm)/:various*',
   },
-  dm: {
-    api_field: 'month',
-    suffix: '/month',
-  },
-}
-
-module.exports = class JitpackDownloads extends BaseJsonService {
-  static get category() {
-    return 'downloads'
-  }
-
-  static get route() {
-    return {
-      base: 'jitpack',
-      pattern:
-        ':interval(dw|dm)/:vcs(github|bitbucket|gitlab|gitee)/:user/:repo',
-    }
-  }
-
-  static get examples() {
-    return [
-      {
-        title: 'JitPack - Downloads',
-        namedParams: {
-          interval: 'dm',
-          vcs: 'github',
-          user: 'jitpack',
-          repo: 'maven-simple',
-        },
-        staticPreview: JitpackDownloads.render({
-          downloads: 14000,
-          interval: 'dm',
-        }),
-        keywords: ['java', 'maven'],
-      },
-    ]
-  }
-
-  static get defaultBadgeData() {
-    return { label: 'downloads' }
-  }
-
-  static render({ downloads, interval }) {
-    return {
-      message: `${metric(downloads)}${intervalMap[interval].suffix}`,
-      color: downloadCount(downloads),
-    }
-  }
-
-  async fetch({ vcs, user, repo }) {
-    return this._requestJson({
-      schema,
-      url: `https://jitpack.io/api/downloads/com.${vcs}.${user}/${repo}`,
-      errorMessages: { 401: 'project not found or private' },
-    })
-  }
-
-  async handle({ interval, vcs, user, repo }) {
-    const json = await this.fetch({ vcs, user, repo })
-    return this.constructor.render({
-      downloads: json[intervalMap[interval].api_field],
-      interval,
-    })
-  }
-}
+  label: 'jitpack',
+  category: 'downloads',
+  dateAdded: new Date('2020-04-05'),
+})

--- a/services/jitpack/jitpack-downloads.tester.js
+++ b/services/jitpack/jitpack-downloads.tester.js
@@ -1,22 +1,23 @@
 'use strict'
 
-const t = (module.exports = require('../tester').createServiceTester())
-const { isMetricOverTimePeriod } = require('../test-validators')
+const { ServiceTester } = require('../tester')
 
-t.create('weekly (github)')
+const t = (module.exports = new ServiceTester({
+  id: 'JitPackDownloads',
+  title: 'JitPackDownloads',
+  pathPrefix: '/jitpack',
+}))
+
+t.create('no longer available (dw)')
   .get('/dw/github/jitpack/maven-simple.json')
-  .timeout(10000)
-  .expectBadge({ label: 'downloads', message: isMetricOverTimePeriod })
-
-t.create('monthly (github)')
-  .get('/dm/github/dcendents/android-maven-gradle-plugin.json')
-  .timeout(10000)
-  .expectBadge({ label: 'downloads', message: isMetricOverTimePeriod })
-
-t.create('unknown package (github)')
-  .get('/dw/github/some-bogus-user/super-fake-project.json')
-  .timeout(10000)
   .expectBadge({
-    label: 'downloads',
-    message: '0/week',
+    label: 'jitpack',
+    message: 'no longer available',
+  })
+
+t.create('no longer available (dm)')
+  .get('/dm/github/jitpack/maven-simple.json')
+  .expectBadge({
+    label: 'jitpack',
+    message: 'no longer available',
   })

